lua/lvim/lsp/init.lua

```diff
@@ -61,20 +61,37 @@ function M.common_on_init(client, bufnr)
 end
 
 function M.common_on_attach(client, bufnr)
+  local MB = 1024 * 1024
+  local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(bufnr))
+  local fsize = ok and stats and stats.size or 0
+
+  if fsize > 5 * MB then
+    vim.lsp.buf_detach_client(bufnr, client.id)
+    Log:info "File larger than 5MB, detached LSP from this buffer"
+    if #vim.lsp.get_buffers_by_client_id(client.id) == 0 then
+      client.stop(true)
+    end
```

Comment:
I don't think that this is needed, and it might even be confusing since you could still create a new file and start editing.

Reply:
> I don't think that this is needed

Stopping the client is in most cases not necessary, only sometimes on very big files jsonls would eat memory until it crashes and that's the only reason for this

>  it might even be confusing since you could still create a new file and start editing.

If you open a new file in a different buffer, the lsp will start automatically.

Reply:
> only sometimes on very big files jsonls would eat memory until it crashes and that's the only reason for this

oof! I think it should be configurable tho, see https://github.com/microsoft/vscode-json-languageservice/blob/fe255ff1dc37e7fcd546a9d9ad82e1495af0ce0c/src/jsonLanguageTypes.ts#L339-L359